src/rust-crypto/libolm_migration.ts

```diff
@@ -380,3 +382,78 @@ async function getAndDecryptCachedSecretKey(
 
     return encodedKey === null ? undefined : await decryptAES(encodedKey, legacyPickleKey, name);
 }
+
+/**
+ * Checks if the legacy store has a trusted public master key.
+ * Returns null if the legacy session was not verified, or the trusted public master key if it was.
+ * @param legacyStore - The legacy store to check.
+ */
+async function getLegacyTrustedPublicMasterKeyBase64(legacyStore: CryptoStore): Promise<string | null> {
+    let maybeTrustedKeys: string | null = null;
+    await legacyStore.doTxn("readonly", "account", (txn) => {
+        legacyStore.getCrossSigningKeys(txn, (keys) => {
+            // can be an empty object after resetting cross-signing keys, see storeTrustedSelfKeys
+            if (keys && Object.keys(keys).length !== 0) {
+                const msk = keys["master"];
+                if (msk) {
+                    // `msk.keys` is an object with { [`ed25519:${pubKey}`]: pubKey }
+                    maybeTrustedKeys = Object.values(msk.keys)[0];
+                }
+            }
+        });
+    });
+
+    return maybeTrustedKeys;
+}
```

Comment:
Done. It's funny because I moved it before thinking you would prefer that.
Can we have a lint rule or something so I don't have to think about order of functions in a file.